src/HasParent.php

```diff
@@ -127,6 +127,11 @@ protected function getParentClass(): string
     {
         static $parentClassName;
 
-        return $parentClassName ?: $parentClassName = (new ReflectionClass($this))->getParentClass()->getName();
+        if ($parentClassName) return $parentClassName;
+
+        $parentClassName = (new ReflectionClass($this))->getParentClass()->getName();
+
+        $parent = new $parentClassName;
+        return $parent->hasParent === true ? $parent->getParentClass() : $parentClassName;
```

Comment:
This may not work as expected.  

Initially `$parentClassName` was set as `static` to cache the value.  If `$parentClassName` had a value, it was returned; otherwise, a value was generated, assigned, then returned.  The next time `getParentClass()` was called, the cached value was returned.

The updated code returns `$parentClassName` if a value exists.
If a value does not exist, one is generated and assigned.
Then we new up the parent class, and check if the parent class has a parent, if it does we call `getParentClass()` and return the result.  The next time this method is called we will return the cached `$parentClassName` and not the result of `$parent->getParentClass()`.

Does that make sense?